modules/azure-firewall/inbound.tf

```
@@ -0,0 +1,43 @@
+data "azurerm_public_ip" "pip" {
+  count               = length(var.firewall)
+  name                = azurerm_public_ip.main[count.index].name
+  resource_group_name = azurerm_public_ip.main[count.index].resource_group_name
+}
+
+resource "azurerm_firewall_nat_rule_collection" "main" {
+  count               = length(var.firewall)
+  name                = "dnatRule-${lookup(element(var.firewall, count.index), "name")}"
+  azure_firewall_name = azurerm_firewall.main[count.index].name
+  resource_group_name = azurerm_resource_group.main[count.index].name
+  priority            = lookup(element(var.firewall, count.index), "priority")
+  action              = "Dnat"
+
+  dynamic "rule" {
+    iterator = rules
+    for_each = lookup(element(var.firewall, count.index), "dnat_rules")
+
+    content {
+      name = "rule-${lookup(element(var.firewall, count.index), "name")}-${rules.value}"
```

do you need the rule prefix?
```suggestion
      name = "${lookup(element(var.firewall, count.index), "name")}-${rules.value}"
```